library/firewall_lib.py

```diff
@@ -251,9 +261,27 @@ def parse_forward_port(module, item):
     return (_port, _protocol, _to_port, _to_addr)
 
 
+def clear_firewalld_settings(fw):
+    configs_to_delete = [
+        "/etc/firewalld/zones/*",
+        "/etc/firewalld/policies/*",
+        "/etc/firewalld/direct.xml",
```

Comment:
One problem we have with the logging role is that, if we remove all of the config files, it is problematic to restore them to the default state.  For example, if you install the `firewalld` RPM package, it installs several files like `/etc/firewalld/firewalld.conf`, `/etc/firewalld/firewalld-server.conf`, etc.  If you use the `firewall` role, does it modify these files?  If so, how do you get back to the original defaults, without having to uninstall and reinstall the `firewalld` RPM package?

Reply:
`/etc/firewalld/firewalld.conf` is an exception here because it comes from RPM and is part of the package. The Role probably should not touch that file.

But the other objects: zones, policies, direct, helpers, icmptypes, ipsets, services - are either non-existent by default or have a "shipped" default in `/usr/lib/firewalld`. They are copied to `/etc/firewalld/` when the user modifies the defaults [0]. As such, deleting them is safe.

[0] example:
```
# firewall-cmd --permanent --zone public --add-service https
```

Reply:
> `/etc/firewalld/firewalld.conf` is an exception here because it comes from RPM and is part of the package. The Role probably should not touch that file.

The role does not explicitly touch that file - but do any of the methods of the `firewall` python library, or objects of type `FirewallClient`, `RichRule`, `FirewallClientZoneSettings`, etc. touch `/etc/firewalld/firewalld.conf`?

> 
> But the other objects: zones, policies, direct, helpers, icmptypes, ipsets, services - are either non-existent by default or have a "shipped" default in `/usr/lib/firewalld`. They are copied to `/etc/firewalld/` when the user modifies the defaults [0]. As such, deleting them is safe.
> 
> [0] example:
> 
> ```
> # firewall-cmd --permanent --zone public --add-service https
> ```

Reply:
> > > `/etc/firewalld/firewalld.conf` is an exception here because it comes from RPM and is part of the package. The Role probably should not touch that file.
> > 
> > 
> > The role does not explicitly touch that file - but do any of the methods of the `firewall` python library, or objects of type `FirewallClient`, `RichRule`, `FirewallClientZoneSettings`, etc. touch `/etc/firewalld/firewalld.conf`?
> 
> Yes. Some of the config knobs in `firewalld.conf` are exposed via CLI (e.g. `--set-log-denied`, `--set-default-zone`) and client library (e.g. FirewallClient: `setLogDenied()`, `setDefaultZone()`).

So how should we handle this?  In the logging role, we had to actually uninstall then reinstall the `rsyslog` package in order to restore `/etc/rsyslog.conf`, but I would rather not do that for the firewall role unless it is absolutely necessary.  I guess the role should make a copy of the original `/etc/firewalld/firewalld.conf` so that it can restore it in this case.

Reply:
RPM won't overwrite `firewalld.conf`. IIRC, it will drop a `firewall.conf.new` next to the user/Ansible modified one. Spec file uses `noreplace`:
```
%config(noreplace) %{_sysconfdir}/firewalld/firewalld.conf
```

Reply:
> RPM won't overwrite `firewalld.conf`. IIRC, it will drop a `firewall.conf.new` next to the user/Ansible modified one. Spec file uses `noreplace`:
> 
> ```
> %config(noreplace) %{_sysconfdir}/firewalld/firewalld.conf
> ```

ok - then 1. uninstall firewalld RPM package 2. remove /etc/firewalld/firewalld.conf 3. install firewalld RPM package

but is this the only way to get back the original /etc/firewalld/firewalld.conf?  If so, then I think the role should make a copy of the original one that it can use to restore.

Reply:
ok - so what should @vrindle do in this case?

Reply:
I think you have to take the rsyslog approach - reinstall the package. I don't like it, but I don't see an alternative.